extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java

```diff
@@ -385,21 +384,33 @@ private ContentTypes() {
         private static final String TEXT_XML = "text/xml";
 
         // WARNING: The order matters for wildcards: if text/json is before text/html, then text/* will match text/json.
-        private static final MIMEHeader[] BASE_HEADERS = {
-                new ParsableMIMEValue(APPLICATION_JSON).forceParse(),
-                new ParsableMIMEValue(TEXT_JSON).forceParse(),
-                new ParsableMIMEValue(TEXT_HTML).forceParse(),
-                new ParsableMIMEValue(APPLICATION_XHTML).forceParse(),
-                new ParsableMIMEValue(APPLICATION_XML).forceParse(),
-                new ParsableMIMEValue(TEXT_XML).forceParse()
-        };
-
-        private static final Collection<MIMEHeader> SUPPORTED = new ArrayList<>(Arrays.asList(BASE_HEADERS));
-        private static final Collection<MIMEHeader> SUPPORTED_CURL = new ArrayList<>();
-        static {
-            SUPPORTED_CURL.add(new ParsableMIMEValue(TEXT_PLAIN).forceParse());
-            SUPPORTED_CURL.addAll(Arrays.asList(BASE_HEADERS));
-            ((ArrayList<MIMEHeader>) SUPPORTED).add(new ParsableMIMEValue(TEXT_PLAIN).forceParse());
+        private static final List<MIMEHeader> BASE_HEADERS = List.of(
+                createParsableMIMEValue(APPLICATION_JSON),
+                createParsableMIMEValue(TEXT_JSON),
+                createParsableMIMEValue(TEXT_HTML),
+                createParsableMIMEValue(APPLICATION_XHTML),
+                createParsableMIMEValue(APPLICATION_XML),
+                createParsableMIMEValue(TEXT_XML));
+
+        private static final Collection<MIMEHeader> SUPPORTED = createSupported();
+        private static final Collection<MIMEHeader> SUPPORTED_CURL = createSupportedCurl();
+
+        private static Collection<MIMEHeader> createSupported() {
+            var supportedCurl = new ArrayList<MIMEHeader>(BASE_HEADERS.size() + 1);
+            supportedCurl.add(createParsableMIMEValue(TEXT_PLAIN));
+            supportedCurl.addAll(BASE_HEADERS);
+            return Collections.unmodifiableCollection(supportedCurl);
+        }
+
+        private static Collection<MIMEHeader> createSupportedCurl() {
+            var supported = new ArrayList<MIMEHeader>(BASE_HEADERS.size() + 1);
+            supported.addAll(BASE_HEADERS);
+            supported.add(createParsableMIMEValue(TEXT_PLAIN));
+            return Collections.unmodifiableCollection(supported);
```

Comment:
Are you sure about the order here ? I would have thought it's the other way around ? (for curl, Text needs to be first)

Reply:
Oops, my bad